client/extensions/zoninator/state/feeds/reducer.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer } from 'state/utils';
+import { ZONINATOR_UPDATE_FEED } from '../action-types';
+
+export const items = createReducer( {}, {
+	[ ZONINATOR_UPDATE_FEED ]: ( state, { siteId, zoneId, data } ) => ( {
+		...state,
+		[ siteId ]: {
+			...state[ siteId ],
+			[ zoneId ]: data,
+		},
+	} ),
+} );
```

Comment:
checkout `keyedReducer()` for a way to focus on the data and not the meta-structure.

```js
export const zoninatorItem = ( state = false, { type, data: { someValue } } ) =>
	ZONINATOR_UPDATE_FEED === type
		? someValue
		: state;

export const zoninatorFeeds = keyedReducer( 'siteId', keyedReducer( 'zoneId', zoninatorItem ) );
```

Also, I'd like to question the implicitness which is hiding _a heap_ of information here. What is the _type_ of `data` and what does it mean? Especially in working with newer reducer code we should be _very explicit_ about what information we are dumping into state. With so many reducers being little more than "dump this thing which isn't described anywhere into state and assume it's all right," we can gain so much more value be being more explicit and treating reducers as a form of a data model. We want to be able to know _why_ `thing.foo` has property `foo` and what we should expect it to contain.